### PR TITLE
Make tzif public API easier to use

### PR DESCRIPTION
**Pull request by nordzilla**

### CODE REVIEW

**Comment by nordzilla** [comment]
> do you also want to add a `categories` key while you're at it?

Ah, I probably should, probably just `["date-and-time", "parser-implementations"]`

I did just publish it, so I'd have to bump to 0.1.1, but that's probably not an issue.

**Review by Manishearth — DISMISSED** [review]

**Comment by nordzilla** [comment]
Hmm, actually, @sffc @Manishearth I want to get your opinion.

I just tried using the crate externally and it's a bit awkward to use. It requires that you also use the `combine` crate to have the `combine` traits available, otherwise you can't parse anything.

There are a couple options:

1) We could re-export combine, and add some sort of `prelude` that people can include that has all the traits they need to parse using `tzif`. I don't really like this, myself.

2) We could do something along the lines of what Shane suggested in (https://github.com/unicode-org/icu4x/pull/2019#discussion_r891820413) and create a `Parser` struct that uses the `combine` traits internally but exposes its own `parse()` function that doesn't require the downstream user to use `combine` directly. 

This seems a lot nicer and easier to use, but it doesn't really give the flexibility of whether you want to use `combine::Parse` or `combine::EasyParse` because there are different actual parsers that you can run over the same implementation.

We could get fancy and expose our own enums or generics to still let users choose between `Parse` and `EasyParse` or something, but for now I think a simple, opinionated all-in-one solution sound the nicest. 

I'd have to bump the version to `0.2.0` but that seems like the best way to go to me.

Sorry, I should have tried using the crate as an external dependency locally before publishing it. Now I know for future crates.

**Comment by nordzilla** [comment]
@sffc @Manishearth I went ahead and implemented option 2) in the comment above. PTAL

File: utils/tzif/src/lib.rs [hunk 1]
```diff
@@ -14,35 +14,53 @@
...
-pub use parse::tzif::tzif;
+/// Error types an implementations.
+pub mod error;
+
+/// A parser for [Time Zone Information Format (`TZif`)](https://tools.ietf.org/id/draft-murchison-tzdist-tzif-00.html) files.
+pub struct TzifParser {}
+
+impl TzifParser {
+    /// Parses a `TZif` file at the provided `path`.
+    pub fn parse_file<P: AsRef<Path>>(path: P) -> Result<TzifData, Error> {
```
Comment (Manishearth) [inline utils/tzif/src/lib.rs:49]: thought: Should these just be toplevel public functions (`parse_tzif_file` and `parse_posix_file`)?

Typically when I see a `Parser` object I imagine a mutable object that holds state

current design is mostly fine though

**Review by Manishearth — DISMISSED** [review]

File: utils/tzif/src/lib.rs (hunk 1 above)
Comment (nordzilla) [reply utils/tzif/src/lib.rs]: Yeah, that's reasonable. 

I'll make the changes.

**Pull request merged**